### PR TITLE
support 'raw' proxy type

### PR DESCRIPTION
**Pull request by aviau**
Hello!

We have a use case where we want to use proxychains with a proxy that does not support CONNECT (some kubernetes-ingress-nginx thigny).

This implementation covers our use case perfectly.

Sending the PR in case you find this useful or if you have feedback.

Thanks for maintaining proxychains :)

Cheers,

### CODE REVIEW

**Comment by rofl0r** [comment]
interesting. how does the "raw" proxy you use decide where to send packets to? is this just a HTTP-only thing ?

btw in case we merge this there would need to be some paragraph added to src/proxychains.conf explaining the proxy type.

**Comment by aviau** [comment]
> is this just a HTTP-only thing ?

In my case it is an HTTP-only thing, but I decided to name it "raw" because it could also be some statically-configured proxy. I am willing to change the name to anything that you prefer.

Example: nginx server dispatching requests to several backends.

I have added a small bit to the docs.

**Comment by aviau** [comment]
Note: This was merged in proxychains here https://github.com/haad/proxychains/pull/108

File: src/core.h [hunk 1]
```diff
@@ -40,6 +40,7 @@ typedef enum {
 
 typedef enum {
 	HTTP_TYPE,
+	RAW_TYPE,
```
Comment (rofl0r) [inline src/core.h:43]: can you move this to last position in the enum, otherwise it might produce weird issues for people forgetting to "make clean" ?

Comment (aviau) [reply src/core.h:43]: done!

**Pull request merged**